src/libraries/System.Text.Json/ref/System.Text.Json.csproj

```diff
@@ -3,6 +3,10 @@
     <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;$(NetFrameworkCurrent)</TargetFrameworks>
     <Nullable>enable</Nullable>
   </PropertyGroup>
+  <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
+  <PropertyGroup>
+    <DefineConstants Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">$(DefineConstants);NETCOREAPPCURRENT</DefineConstants>
```

Comment:
You don't need to do this yourself. The SDK automatically creates define constants for each TFM. And it also creates a "version-less" define as well.

So instead, you can delete these lines, and in the .cs file you can say: `#if NETCOREAPP`